File: llm_generated_code/Gemini/CPP/Level2/Q32.cpp
#include <iostream>
#include <string>

using namespace std;

bool isMeow(const string& s) {
    int n = s.size();
    if (n < 4) {
        return false;
    }

    int i = 0;
    while (i < n && (s[i] == 'm' || s[i] == 'M')) {
        i++;
    }

    if (i == 0 || i == n) {
        return false;
    }

    while (i < n && (s[i] == 'e' || s[i] == 'E')) {
        i++;
    }

    if (i == 0 || i == n) {
        return false;
    }

    while (i < n && (s[i] == 'o' || s[i] == 'O')) {
        i++;
    }

    if (i == 0 || i == n) {
        return false;
    }

    while (i < n && (s[i] == 'w' || s[i] == 'W')) {
        i++;
    }

    return i == n;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        string s;
        cin >> s;

        if (isMeow(s)) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }

    return 0;
}

File: llm_generated_code/Gemini/Python/Level2/Q32.py
def is_meow(s):
    if len(s) < 4:
        return False
    
    i = 0
    while i < len(s) and (s[i] == 'm' or s[i] == 'M'):
        i += 1
    if i == 0 or i == len(s):
        return False
    
    while i < len(s) and (s[i] == 'e' or s[i] == 'E'):
        i += 1
    if i == 0 or i == len(s):
        return False
    
    while i < len(s) and (s[i] == 'o' or s[i] == 'O'):
        i += 1
    if i == 0 or i == len(s):
        return False
    
    while i < len(s) and (s[i] == 'w' or s[i] == 'W'):
        i += 1
    return i == len(s)

def main():
    t = int(input())
    for _ in range(t):
        s = input()
        print("YES" if is_meow(s) else "NO")

if __name__ == "__main__":
    main()
